File: tests/positives/framework-detection/dart/detect-gemini.dart
// Positive test cases for Gemini (Dart)

// ruleid: detect-gemini
import 'package:google_generative_ai/google_generative_ai.dart';

void main() async {
  // ruleid: detect-gemini
  final model = GenerativeModel(model: 'gemini-pro', apiKey: 'key');

  // ruleid: detect-gemini
  final response = await model.generateContent([Content.text('Hello')]);

  // ruleid: detect-gemini
  final chat = model.startChat();
}

File: tests/positives/framework-detection/swift/detect-gemini.swift
// Positive test cases for Gemini (Swift)

// ruleid: detect-gemini
import GoogleGenerativeAI

// ruleid: detect-gemini
let model = GenerativeModel(name: "gemini-pro", apiKey: "key")

// ruleid: detect-gemini
let response = try await model.generateContent("Hello")

// ruleid: detect-gemini
let chat = model.startChat()
